types/ramda/index.d.ts

```diff
@@ -1858,6 +1885,35 @@ export function of<T>(x: T): T[];
 export function omit<T, K extends string>(names: readonly K[], obj: T): Omit<T, K>;
 export function omit<K extends string>(names: readonly K[]): <T>(obj: T) => Omit<T, K>;
 
+/**
+ * Takes a binary function `f`, a unary function `g`, and two values.
+ * Applies `g` to each value, then applies the result of each to `f`.
+ * Also known as the P combinator.
+ *
+ * @example
+ * ```typescript
+ * const eqBy = R.on((a, b) => a === b);
+ * eqBy(R.prop('a'), {b:0, a:1}, {a:1}) //=> true;
+ *
+ * const containsInsensitive = R.on(R.contains, R.toLower);
+ * containsInsensitive('o', 'FOO'); //=> true
+ * ```
+ */
+export function on<A, B, C>(f: (a: A, b: A) => B, g: (a: C) => A, a: C, b: C): B;
```

Comment:
Maybe we can change the generic parameters order to :

```ts
export function on<A, B, C>(f: (a: B, b: B) => C, g: (a: A) => B, a: A, b: A): C;
```

`A` is input type, `B` is intermediate type, and `C` is output type.

Reply:
right, I think renaming to `T`, `U`, and `R` might be good too?
also, I wonder if the overloads where `a` and `b` are provided separately should be removed. or at least these two:
```typescript
export function on<A, B, C>(f: (a: A, b: A) => B, g: (a: C) => A, a: C): (b: C) => B;
<T>(g: (a: T) => U, a: T): (b: T) => R;
```